src/xdp-utils.c

```diff
@@ -2151,7 +2151,21 @@ app_info_map_pids (XdpAppInfo  *app_info,
   g_return_val_if_fail (app_info != NULL, FALSE);
   g_return_val_if_fail (pids != NULL, FALSE);
 
-  if (app_info->kind != XDP_APP_INFO_KIND_FLATPAK)
+  if (app_info->kind == XDP_APP_INFO_KIND_SNAP)
+    {
+      for (int i = 0; i < n_pids; i++)
+        {
+          XdpAppInfo *info = xdp_get_app_info_from_pid (pids[i], error);
+          if (!info || info->kind != app_info->kind || g_strcmp0 (info->id, app_info->id) != 0)
+            {
+              g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_PERMISSION_DENIED,
+                                   "App info of pids do not match.");
```

Comment:
```suggestion
          if (!info)
            return FALSE;
   
          if (info->kind != app_info->kind || g_strcmp0 (info->id, app_info->id) != 0)
            {
              g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_PERMISSION_DENIED,
                                   "App info of pids do not match.");
```

In case in the `(!info)` case (when also `error` is set). you can use `g_propagate_prefixed_error`.

Reply:
```suggestion
          g_autoptr(GError) local_error = NULL;
          XdpAppInfo *info = xdp_get_app_info_from_pid (pids[i], &local_error);
          if (!info)
            {
              g_propagate_prefixed_error(error, g_steal_pointer(&local_error), "Invalid app");
              return FALSE;
            }

          if (info->kind != app_info->kind || g_strcmp0 (info->id, app_info->id) != 0)
            {
              g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_PERMISSION_DENIED,
                                   "App info of pids do not match.");
              return FALSE;
            }
```

Something like this?

Reply:
Yes, a part from using correct spacing `g_propagate_prefixed_error (error, g_steal_pointer (&local_error), "Can't find app for pid %" G_PID_FORMAT ": ", pid[i]);`

Reply:
In general though, you can also decide to skip the error: do you want all the pids to be mapped, right?

Reply:
Yeah, each pid just needs to be confirmed to belong to the Snap

Reply:
Fixed in https://github.com/flatpak/xdg-desktop-portal/commit/3055e448bd61dd551be6fce48d15ce70751182f6